libs/cases/backend/case-management/src/lib/services/case.service.ts

```diff
@@ -21,14 +29,86 @@ export class XdCaseService {
 	public getAllCases(): Observable<ICaseResponse[]> {
 		return from(this.prismaService.case.findMany()).pipe(
 			map((items) =>
-				items.map((item) => ({
-					...item,
-					overdue: Date.now() > new Date(item.dueDate).getTime(),
-				}) as ICaseResponse),
+				items.map((item) => {
+					const { assetAssetId, type, status, priority, ...rest } = item;
+
+					return {
+						...rest,
+						type: type as unknown as ECaseType,
+						status: status as unknown as ECaseStatus,
+						priority: priority as unknown as ECasePriority,
```

Comment:
I have to say the  `type as unknown as ECaseType` but ther probably is no better way, right?